### PR TITLE
Deserialize `java.util.UUID` encoded as Base64 and base64Url with or without padding

### PR DESCRIPTION
**Pull request by jebl01**
*Vert.x 4* changed to `Base64UrlEncoder` as the default for binary columns in database responses (*Vert.x* serializes database responses into JSON arrays). This becomes a problem when Jackson tries to deserialize into a field of type `UUID`. The workaround is to provide a custom serializer for UUID or to have `byte[]` fields in your database entities.

This PR ensures all variants of Base64 is supported for UUIDs "out of the box".

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java:46]: Isn't this remarkably inefficient way to do this, using dynamically compiled regexp replaced and constructing Strings on the fly? Should be easy enough to change to character-based alternative.

But overall I am not sure how safe it is to start replacing characters here. I'll have to think about this a bit.

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java:52]: That's a good point, makes sense.

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java
Comment (yawkat) [reply src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java:46]: `replace(String, String)` is actually not regex, and it even delegates to `replace(char, char)` if both strings are length 1. still should move to the char version though.

Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java:46]: Ah ok. I misremembered that then. Should probably move logic to separate helper method regardless.

File: src/test/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializerTest.java
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializerTest.java:14]: Instead of using new generated random UUIDs, it'd be better to generate one randomly but include as constant.
This to avoid any possibility of having non-reproducible fails (at expense of not covering many combinations).

It is unfortunate JDK does not offer a way to see generator for random-number variant, for testing (but perhaps they were worried about devs using that in production and end up with collisions), as otherwise we'd use that.
JUG of course can do that but perhaps it's better not to add it as a (test) dependency,.

File: src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java
Comment (jebl01) [reply src/main/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializer.java]: Ok, changed to chars for the replacement now + separated the logic to helper methods.

File: src/test/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializerTest.java
Comment (jebl01) [reply src/test/java/com/fasterxml/jackson/databind/deser/std/UUIDDeserializerTest.java]: ok, I can do that 👍

**Comment by cowtowncoder** [comment]
@jebl01 Ok, looks good -- I think this makes sense and timing is good as this can make it into 2.17.0-rc1 release, to be released soon (within a week I hope).

One thing I'd need before final review & merge: CLA (if we haven't already gotten one -- apologies if I missed earlier one).
It's from 

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

(there's alternative CCLA too if preferable)

and is usually easiest to print, fill & sign, scan/photo, email to `cla` at fasterxml dot com.
Once we get it I can proceed with merging and we can the improvement in.

Thank you once again for contributing this!

**Comment by jebl01** [comment]
CLA in the mail 📫

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
Looks good, made some non-functional changes to help merging to `master` (3.0).

**Pull request merged**
